### PR TITLE
chore(replay): add a log sampling filter and sample replay ingest logs

### PR DESCRIPTION
**Pull request by aliu39**
Closes
- https://github.com/getsentry/team-replay/issues/495

### CODE REVIEW

File: src/sentry/replays/usecases/ingest/__init__.py [hunk 1]
```diff
@@ -24,7 +25,18 @@
 from sentry.utils import json, metrics
 from sentry.utils.outcomes import Outcome, track_outcome
 
+MOBILE_EVENT_SAMPLE_RATE = 0.5
```
Comment (aliu39) [inline src/sentry/replays/usecases/ingest/__init__.py:28]: Can make these options if we want. Values right now are placeholders

File: src/sentry/logging/handlers.py [hunk 2]
```diff
@@ -149,3 +150,22 @@ def emit(self, record, logger=None):
...
+    """
+    A logging filter to sample messages at different rates.
+
+    prob_mapping -- a mapping of messages to probabilities. The msg must be an exact match.
+      Messages not in the mapping aren't filtered.
+    """
+
+    def __init__(self, prob_mapping: dict[str, float]):
+        super().__init__()
+        self.prob_mapping = prob_mapping
```
Comment (cmanallen) [inline src/sentry/logging/handlers.py:165]: Don't abbreviate.  `probability_map` or `probability_mapping`.

File: src/sentry/replays/consumers/recording_buffered.py [hunk 3]
```diff
@@ -72,6 +77,13 @@
 from sentry.utils import json, metrics
 
 logger = logging.getLogger(__name__)
+logger.addFilter(
+    SamplingFilter(
+        {
+            "mobile_event": MOBILE_EVENT_SAMPLE_RATE,
```
Comment (cmanallen) [inline src/sentry/replays/consumers/recording_buffered.py:83]: Why is this different from the other consumer?

File: src/sentry/replays/usecases/ingest/__init__.py [hunk 4]
```diff
@@ -24,7 +25,18 @@
...
+RRWEB_EVENT_COUNT_SAMPLE_RATE = 0.5
+
 logger = logging.getLogger("sentry.replays")
+logger.addFilter(
+    SamplingFilter(
+        {
+            "mobile_event": MOBILE_EVENT_SAMPLE_RATE,
+            "rrweb_event_count": RRWEB_EVENT_COUNT_SAMPLE_RATE,
```
Comment (cmanallen) [inline src/sentry/replays/usecases/ingest/__init__.py:36]: Are either of these log messages emitted in our consumer?

File: src/sentry/replays/usecases/ingest/__init__.py (hunk 1 above)
Comment (cmanallen) [inline src/sentry/replays/usecases/ingest/__init__.py:28]: Let's make this an option.

**Review by cmanallen — CHANGES_REQUESTED** [review]

File: src/sentry/replays/usecases/ingest/__init__.py (hunk 4 above)
Comment (aliu39) [reply src/sentry/replays/usecases/ingest/__init__.py:36]: Do you mean a different file?

File: src/sentry/replays/consumers/recording_buffered.py (hunk 3 above)
Comment (aliu39) [reply src/sentry/replays/consumers/recording_buffered.py:83]: Are you suggesting we add rrweb_event_count here too? It's not currently logged in this file

Comment (cmanallen) [reply src/sentry/replays/consumers/recording_buffered.py:83]: No just asking the question.  If we're not logging it we don't need to include it.

File: src/sentry/replays/usecases/ingest/__init__.py (hunk 4 above)
Comment (aliu39) [reply src/sentry/replays/usecases/ingest/__init__.py:36]: Just realized I was looking at the `event_type` in `extras`. The msg is `sentry.replays.slow_click`. I'll have to rethink this

Comment (aliu39) [reply src/sentry/replays/usecases/ingest/__init__.py:36]: Do we want different rates for these, or could we sample all slow click logs at the same rate for now?

Comment (cmanallen) [reply src/sentry/replays/usecases/ingest/__init__.py:36]: Same rate is fine.

File: src/sentry/replays/consumers/recording_buffered.py (hunk 3 above)
Comment (aliu39) [reply src/sentry/replays/consumers/recording_buffered.py:83]: We're only logging mobile_event in this one

File: src/sentry/replays/usecases/ingest/__init__.py (hunk 4 above)
Comment (cmanallen) [reply src/sentry/replays/usecases/ingest/__init__.py:36]: Can you access the `extra` field within your sampler class?

Comment (aliu39) [reply src/sentry/replays/usecases/ingest/__init__.py:36]: I can. The filter will become specific to our usecase though, so I'd move it into the __init__ file

**Comment by JoshFerge** [comment]
@cmanallen @aliu39 need any help w/ this? :D

**Comment by cmanallen** [comment]
@JoshFerge Andrew and I are on PTO until next week.  You're welcome to help out!  I'll take a peak at this tomorrow or Wednesday and review it for you (if you make any changes)!

**Comment by aliu39** [comment]
IIRC here was a reason Colton blocked this that we forgot to document here. Last we talked we were thinking of trying a whole different approach

**Comment by aliu39** [comment]
cc @cmanallen 
Sorry - I completely forgot our discussion 😥. 
> i was thinking we could just sample 1% of logs. is there any reason that doesn't work?

@JoshFerge Looking at this now that seems fine to me. FF is launching tomorrow, I'll follow up after and try to close this by the end of the week.

**Comment by aliu39** [comment]
Sorry this got lost.. since this is tracked in the ticket, I'll close, and we can revisit after holidays

**Pull request closed without merging**